client/core/trade.go

```diff
@@ -2600,14 +2600,30 @@ func (c *Core) sendInitAsync(t *trackedTrade, match *matchTracker, coinID, contr
 		err = t.dc.signAndRequest(init, msgjson.InitRoute, ack, timeout)
 		if err != nil {
 			var msgErr *msgjson.Error
-			if errors.As(err, &msgErr) && msgErr.Code == msgjson.SettlementSequenceError {
-				c.log.Errorf("Starting match status resolution for 'init' request SettlementSequenceError")
-				c.resolveMatchConflicts(t.dc, map[order.OrderID]*matchStatusConflict{
-					t.ID(): {
-						trade:   t,
-						matches: []*matchTracker{match},
-					},
-				})
+			if errors.As(err, &msgErr) {
+				if msgErr.Code == msgjson.SettlementSequenceError {
+					c.log.Errorf("Starting match status resolution for 'init' request SettlementSequenceError")
+					c.resolveMatchConflicts(t.dc, map[order.OrderID]*matchStatusConflict{
+						t.ID(): {
+							trade:   t,
+							matches: []*matchTracker{match},
+						},
+					})
+				} else if msgErr.Code == msgjson.RPCUnknownMatch {
+					oid := t.ID()
+					c.log.Warnf("DEX %s did not report active match %s on order %s - assuming revoked, status %v.",
+						t.dc.acct.host, match, oid, match.Status)
+					// We must have missed the revoke notification. Flag to allow recovery
+					// and subsequent retirement of the match and parent trade.
+					match.MetaData.Proof.SelfRevoked = true
```

Comment:
Don't you need to lock the `trackedTrade.mtx` to modify the `Proof`?

Reply:
yes https://github.com/decred/dcrdex/compare/eef6d19a4c715c7ef42c2313756879851f0d40e2..e6533b691c41d3939fb293c3d1572154e1679b20

Reply:
Did we need to hold it to read `match.Status`?

Reply:
Moved it up https://github.com/decred/dcrdex/compare/e6533b691c41d3939fb293c3d1572154e1679b20..377603af2b79e07c5347ce5b3da752d3b126e5e9